app/src/main/java/org/mozilla/tv/firefox/FirefoxApplication.kt

```diff
@@ -50,6 +51,17 @@ open class FirefoxApplication : LocaleAwareApplication() {
             // Enable crash reporting. Don't add anything above here because if it crashes, we won't know.
             SentryIntegration.init(this, serviceLocator.settingsRepo)
 
+            // Set up Glean by connecting it to the preference flag for data collection and calling
+            // the initialize() function.
+            with(serviceLocator.settingsRepo) {
+                dataCollectionEnabled.observeForever { collectionEnabled ->
+                    if (collectionEnabled != null) {
+                        Glean.setUploadEnabled(collectionEnabled)
+                    }
+                }
+            }
+            Glean.initialize(applicationContext)
```

Comment:
Nit: This function operates at a lower level of abstraction than the rest of `onCreate`, so I'd rather move it into a helper function.